mycore-base/src/main/java/org/mycore/access/mcrimpl/MCRUserClause.java

```diff
@@ -66,11 +66,20 @@ private Pattern toRegex(String userExp) {
     }
 
     public boolean evaluate(Object o) {
-        if (userRegEx != null) {
-            return userRegEx.matcher(MCRSessionMgr.getCurrentSession().getUserInformation().getUserID()).matches()
-                ^ not;
-        }
-        return user.equals(MCRSessionMgr.getCurrentSession().getUserInformation().getUserID()) ^ not;
+    	if(o == null || !(o instanceof MCRAccessData) || ((MCRAccessData)o).getUserInformation() == null) {
```

Comment:
Could be simplified to:
```
        final MCRUserInformation userInformation = Optional.ofNullable(o)
            .filter(obj -> obj instanceof MCRAccessManager)
            .map(MCRAccessData.class::cast)
            .map(MCRAccessData::getUserInformation)
            .orElseGet(MCRSessionMgr.getCurrentSession()::getUserInformation);

        if (userRegEx != null) {
            return userRegEx.matcher(userInformation.getUserID()).matches() ^ not; 
        }
        return user.equals(userInformation.getUserID()) ^ not;
```

Reply:
should be: .filter(obj -> obj instanceof MCRAccessData)